src/PubSub/Drivers/RedisClient.php

```diff
@@ -258,6 +288,17 @@ public function channelMemberCounts($appId, array $channelNames): PromiseInterfa
             });
     }
 
+    /**
+     * Get the amount of connections aggregated on multiple instances.
+     *
+     * @param  mixed  $appId
+     * @return null|int|\React\Promise\PromiseInterface
+     */
+    public function getGlobalConnectionsCount($appId)
+    {
+        return $this->publishClient->hget($this->getTopicName($appId), 'connections');
```

Comment:
@francislavoie I'm not sure how to resolve this promise. 😢

Reply:
https://github.com/beyondcode/laravel-websockets/blob/2.x/src/Statistics/Logger/MemoryStatisticsLogger.php#L108-L110

Just change this to:

```php
$currentConnectionCount = $this->channelManager->getConnectionCount($appId);

if ($currentConnectionCount instanceof PromiseInterface) {
	$currentConnectionCount->then(function($currentConnectionCount) use ($statistic) {
		$statistic->reset($currentConnectionCount);
	}, function () {
		// optional error handling
	});
} else {
	$statistic->reset($currentConnectionCount);
}
```

Compare with how it's handled in the HTTP controller:

https://github.com/beyondcode/laravel-websockets/blob/2.x/src/HttpApi/Controllers/Controller.php#L193-L205

Reply:
Ooh, now it makes sense. 🤦

Reply:
Also in https://github.com/beyondcode/laravel-websockets/blob/4067432b933b435fca96e64924fd22ce22792803/src/Console/StartWebSocketServer.php#L113

I think you should instead do:

```php
$statsLogger = $this->app->get(StatisticsDriver::class);
$statsLogger->save();
```

Seems weird to use a facade for that.

Reply:
Man, you are a genius. 🚀 I just now need to add some functional tests to catch all of the nasty stuff and perhaps separate the statistics dump into a separate command that can be cron'd to avoid cache locking that blocks the stuff. 🤔